dissect/target/plugins/os/windows/rdpcache.py

```diff
@@ -0,0 +1,478 @@
+                64 - tile_header.tile_width,
+                64 - tile_header.tile_height,
+                remnant_tile_colors,
+                is_remnant=True,
+            )
+
+
+class RdpCachePlugin(Plugin):
+    """Plugin to extract the RDP Bitmap Cache from a Windows target.
+
+    Resources:
+        - https://www.cert.ssi.gouv.fr/actualite/CERTFR-2016-ACT-017/
+    """
+
+    __namespace__ = "rdpcache"
+
+    CACHE_PATH = "AppData/Local/Microsoft/Terminal Server Client/Cache/"
+    GLOBS = ["Cache*.bin", "bcache2*.bmc"]
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._caches: list[tuple[UserDetails, TargetPath]] = []
+
+        for user_details in self.target.user_details.all_with_home():
+            user_rdp_cache = user_details.home_path.joinpath(self.CACHE_PATH)
+            for glob in self.GLOBS:
+                for cache_path in user_rdp_cache.glob(glob):
+                    if cache_path.lstat().st_size > 0:
+                        self._caches.append((user_details, cache_path))
+
+    def check_compatible(self) -> None:
+        """At least one bitmap cache file with contents is necessary."""
+        if not self._caches:
+            raise UnsupportedPluginError("No RDP Cache files found on target")
+
+    def _recover_tiles_from_cache(self, path: TargetPath) -> Iterator[BitmapTile]:
+        """Given a target path, select the correct extraction method and yield bitmap tiles."""
+        fh = path.open("rb")
+        if fh.read(len(BIN_MAGIC)) == BIN_MAGIC:
+            yield from extract_bin(fh)
+        else:
+            yield from extract_bmc(fh)
+
+    @export(record=RDPCacheRecord)
+    def paths(self) -> Iterator[RDPCacheRecord]:
+        """Yield paths and timestamps of RDP Cache bitmap files."""
+        for user_details, path in self._caches:
+            lstat = path.lstat()
+            yield RDPCacheRecord(
+                ctime=lstat.st_ctime,
+                mtime=lstat.st_mtime,
+                path=path,
+                _user=user_details.user,
+                _target=self.target,
+            )
+
+    @arg("--output", "-o", dest="output_dir", type=Path, required=True, help="path to recover bitmap files to")
+    @arg(
+        "--grid",
+        "-g",
+        dest="as_grid",
+        action="store_true",
+        help="assemble all recovered tiles of a cache into a single grid image with borders around each tile",
+    )
+    @arg(
+        "--collage",
+        "-c",
+        dest="as_collage",
+        action="store_true",
+        help="assemble all recovered tiles of a cache into one image",
+    )
+    @arg(
+        "--no-tiles",
+        "-n",
+        dest="no_individual_tiles",
+        action="store_true",
+        help="do not save each recovered tile as a separate bitmap",
+    )
+    @arg(
+        "--remnants",
+        "-r",
+        choices=["only", "include", "exclude"],
+        default="exclude",
+        type="str",
```

Comment:
```suggestion
        type=str,
```
This actually causes a failure when trying to execute the plugin. Something to unit test in the future 😄.